### ISSUE TITLE
tld: Expand data sources

### ISSUE THREAD

**Issue by dgw** [sopel-irc/sopel#1648]
The `tld` plugin is not as useful as it once was. Queries for TLDs that definitely exist often fail, and successful queries return incomplete data much of the time.

Relying on Wikipedia is probably not our best long-term option…

https://github.com/sopel-irc/sopel/blob/b105fe4aaa6c1cd258337e60a4f17c1a0751ecb5/sopel/modules/tld.py#L23

At one point I was toying with the idea of adding [ICANNWiki](https://icannwiki.org/) as a source, or replacing Wikipedia entirely. I lost interest in that specific idea when I realized that ICANNWiki actually lacks _more_ details for many of its listed TLDs than Wikipedia usually does—but I still think we should try to improve the output of `.tld` commands.

Hence, this feature issue. It's not yet tied to a version milestone, and I would love to hear suggestions for other data sources that we might use for the `.tld` command besides our one Wikipedia page listing a bunch of TLDs (but not all of them).

**Comment by kwaaak** [comment]
The module relies a lot on editorial content on Wikipedia.

We could use [this IANA text file](https://data.iana.org/TLD/tlds-alpha-by-domain.txt) for searching and at least give a positive result if the domain exists.
If there is more information available on Wikipedia (or other sources), display that as well.

There is no native python whois module, we could use [IANA Whois](https://www.iana.org/whois?q=net) for basic information like creation date and status.

**Comment by TigerP** [comment]
You could also try using [RDAP](https://www.icann.org/rdap), which should be available for all gTLD's from august 26th 2019. But probably not for ccTLD's.

**Comment by dgw** [comment]
I'm putting the finishing touches on a rewrite of this plugin. Whether it's enough to consider this issue closed is undecided yet, but it is a massive improvement nonetheless.

Sneak peek output:

```irc
<SopelTest> [tld] Name: .aq | Entity: Antarctica | IDN: ? | DNSSEC: No | SLD: Yes | IPv6: ? | Notes: Defined by
            the Antarctic Treaty as everything south of latitude 60°S. AQ domain names are available to government […]
```

**Comment by dgw** [comment]
Considering this "fixed" by #1939. With IANA's list for checking existence (thanks to @kwaaak's above suggestion) and better (dynamic!) table parsing off of Wikipedia for the details, this feature should be secure Enough™ until and unless a proper API for this kind of data can be found.
